go/vt/discovery/healthcheck.go

```diff
@@ -63,25 +63,33 @@ type HealthCheck interface {
 }
 
 // NewHealthCheck creates a new HealthCheck object.
-func NewHealthCheck(connTimeout time.Duration, retryDelay time.Duration) HealthCheck {
+func NewHealthCheck(connTimeout time.Duration, retryDelay time.Duration, healthCheckTimeout time.Duration) HealthCheck {
 	hc := &HealthCheckImpl{
-		addrToConns: make(map[string]*healthCheckConn),
-		targetToEPs: make(map[string]map[string]map[topodatapb.TabletType][]*topodatapb.EndPoint),
-		connTimeout: connTimeout,
-		retryDelay:  retryDelay,
+		addrToConns:        make(map[string]*healthCheckConn),
+		targetToEPs:        make(map[string]map[string]map[topodatapb.TabletType][]*topodatapb.EndPoint),
+		connTimeout:        connTimeout,
+		retryDelay:         retryDelay,
+		healthCheckTimeout: healthCheckTimeout,
 	}
 	if hcConnCounters == nil {
 		hcConnCounters = stats.NewMultiCountersFunc("HealthcheckConnections", []string{"keyspace", "shardname", "tablettype"}, hc.servingConnStats)
 	}
+	go func() {
+		t := time.NewTicker(healthCheckTimeout / 3)
+		for range t.C {
+			hc.checkHealthCheckTimeout()
+		}
```

Comment:
Added healthcheck.Close().
I tried to change Stop() in binlog.go, but it causes some problems like multiple calls to healthcheck.Close(), and also deadlock in unit test for tabletmanager.